Create number-format.md, closes #3219

reganmarie opened the pull request:
- Created number-format.md on filepath docs/content/php/concepts/string-functions/terms/number-format/number-format.md
- Added Title, Description, Subjects, Tags, and Catalog Content
- Added description, syntax, example, and codebyte example
- Ran through 3 online plagiarism checkers to ensure 100% unique wording and description, passed with 100%
- Followed content and style guidelines -Checked for spelling and grammar errors

<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

caupolicandiaz commented on content/php/concepts/string-functions/terms/number-format/number-format.md:
```diff
@@ -0,0 +1,54 @@
+---
+Title: 'number_format()'
+Description: 'Formats a number with grouped thousands'
```
```suggestion
Description: 'Returns a given value formatted with grouped thousands.'
```

caupolicandiaz commented on content/php/concepts/string-functions/terms/number-format/number-format.md:
```diff
@@ -0,0 +1,54 @@
+Subjects:
+  - 'Computer Science'
+  - 'Web Design'
+  - 'Web Development'
```
```suggestion
```

caupolicandiaz replied:
there should only be two items under subjects

caupolicandiaz commented on content/php/concepts/string-functions/terms/number-format/number-format.md:
```diff
@@ -0,0 +1,54 @@
+
+- `number`: The number to format.
+- `decimals`: Optional argument that sets the number of digits after the decimal point. If not given, the number returned will be a whole number.
+- `decimal_point`: Optional argument that specifies the decimal separator and should be given in string format.
```
```suggestion
- `decimal_point`: Optional parameter that specifies the decimal separator and should be given as a string.
```

caupolicandiaz commented on content/php/concepts/string-functions/terms/number-format/number-format.md:
```diff
@@ -0,0 +1,54 @@
+- `number`: The number to format.
+- `decimals`: Optional argument that sets the number of digits after the decimal point. If not given, the number returned will be a whole number.
+- `decimal_point`: Optional argument that specifies the decimal separator and should be given in string format.
+- `separator`: Optional argument that specifies the thousands separator and should be given in string format.
```
```suggestion
- `separator`: Optional parameter that specifies the thousands separator and should be given as a string.
```

caupolicandiaz commented on content/php/concepts/string-functions/terms/number-format/number-format.md:
````diff
@@ -0,0 +1,54 @@
+?>
+```
+
+This is the output for the above example:
````
```suggestion
This will result in the following output:
```

caupolicandiaz commented on content/php/concepts/string-functions/terms/number-format/number-format.md:
````diff
@@ -0,0 +1,54 @@
+
+```codebyte/php
+<?php
+  echo number_format("500000");
````
```suggestion
  echo number_format("500000"), "\n";
```

caupolicandiaz commented on content/php/concepts/string-functions/terms/number-format/number-format.md:
```diff
@@ -0,0 +1,54 @@
+
+## Codebyte Example
+
+The following codebyte example can be run and calls the `number_format()` method two times. The second call uses the optional `decimals`, `decimal_point`, and `separator` arguments.
```
```suggestion
The following codebyte example can be run and calls the `number_format()` method two times. The second call uses the optional `decimals`, `decimal_point`, and `separator` parameters.
```

caupolicandiaz reviewed (changes requested):
@reganmarie I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

reganmarie commented:
@caupolicandiaz I didn't want to spam your inbox by responding to each individual revision, so these are the changes I made in the latest commit!
- Changed Description to 'Returns a given value formatted with grouped thousands.'
- Removed a subject so there are only two
- Changed argument to parameter in syntax
- Changed "given in string format" to "given as a string" in both instances that wording was used in Syntax block
- Changed output wording to "this will result in the following output" in example section
- Added new line to Codebyte example
- Changed 'arguments' to 'parameters' in Codebyte explanation

caupolicandiaz reviewed (approved):
@reganmarie This is ready for a second review.

reganmarie commented:
Thank you @caupolicandiaz 😊

SSwiniarski reviewed (approved):
@reganmarie Looks good.
